src/hiccup/form.clj

```diff
@@ -74,6 +74,19 @@
              :value value
              :checked checked?}]))
 
+;; According to the HTML spec, the value inside the boolean `selected` attribute
+;; can only be "" or "selected" (case insensitive).  Hence, the coercion to
+;; boolean is required (otherwise, the attribute value will be `x` itself).
+;; We also need the `(not (sequential? selected))` requirement in the
+;; second `and` because if `selected` is a vector where `x` is not an element,
+;; the first `and` will fail, and the second will try to run something
+;; like `([a b] c)`, and can easily lead to java.lang.IndexOutOfBoundsException.
```

Comment:
This comment can be part of the pull request. I don't think it needs to be embedded in the code as it's not going to be relevant information for most people reading it.

Reply:
Got it.  Will remove.